File: app/src/main/java/nz/co/codebros/quakesnz/ui/InfoActivity.java
package nz.co.codebros.quakesnz.ui;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.ViewPager;
import android.support.v7.app.ActionBar;
import android.support.v7.app.ActionBarActivity;

import com.google.android.gms.analytics.GoogleAnalytics;

import nz.co.codebros.quakenz.R;
import nz.co.codebros.quakesnz.QuakesNZApplication;

/**
 * Created by Leandro on 24/07/2014.
 */
public class InfoActivity extends ActionBarActivity implements ActionBar.TabListener, ViewPager.OnPageChangeListener {

    private ViewPager mViewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((QuakesNZApplication) getApplication())
                .getTracker(QuakesNZApplication.TrackerName.APP_TRACKER);

        setContentView(R.layout.activity_info);

        mViewPager = (ViewPager) findViewById(R.id.pager);
        mViewPager.setAdapter(new FragmentStatePagerAdapter(getSupportFragmentManager()) {

            @Override
            public Fragment getItem(int i) {
                int title, body;

                switch (i) {
                    case 0:
                        title = R.string.before_title;
                        body = R.string.before_body;
                        break;
                    case 1:
                        title = R.string.during_title;
                        body = R.string.during_body;
                        break;
                    case 2:
                    default:
                        title = R.string.after_title;
                        body = R.string.after_body;
                }

                return InfoFragment.newInstance(title, body);
            }

            @Override
            public int getCount() {
                return 3;
            }
        });

        mViewPager.setOnPageChangeListener(this);

        final ActionBar actionBar = getSupportActionBar();

        actionBar.setNavigationMode(ActionBar.NAVIGATION_MODE_TABS);
        actionBar.addTab(actionBar.newTab().setText(R.string.before).setTabListener(this));
        actionBar.addTab(actionBar.newTab().setText(R.string.during).setTabListener(this));
        actionBar.addTab(actionBar.newTab().setText(R.string.after).setTabListener(this));
    }

    @Override
    public void onTabSelected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {
        mViewPager.setCurrentItem(tab.getPosition());
    }

    @Override
    public void onTabUnselected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {

    }

    @Override
    public void onTabReselected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {

    }

    @Override
    public void onPageScrolled(int i, float v, int i2) {

    }

    @Override
    public void onPageSelected(int i) {
        getSupportActionBar().setSelectedNavigationItem(i);
    }

    @Override
    public void onPageScrollStateChanged(int i) {

    }

    @Override
    protected void onStart() {
        super.onStart();
        GoogleAnalytics.getInstance(this).reportActivityStart(this);
    }

    @Override
    protected void onStop() {
        super.onStop();
        GoogleAnalytics.getInstance(this).reportActivityStop(this);
    }
}

File: app/src/main/java/nz/co/codebros/quakesnz/ui/InfoActivity.kt
package nz.co.codebros.quakesnz.ui

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.app.FragmentTransaction
import android.support.v4.view.ViewPager
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity

import nz.co.codebros.quakesnz.R

/**
 * Created by Leandro on 24/07/2014.
 */
class InfoActivity : AppCompatActivity(), ActionBar.TabListener, ViewPager.OnPageChangeListener {

    private var mViewPager: ViewPager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_info)

        mViewPager = findViewById<ViewPager>(R.id.pager)
        mViewPager?.adapter = object : FragmentStatePagerAdapter(supportFragmentManager) {

            override fun getItem(i: Int): Fragment {
                val title: Int
                val body: Int

                when (i) {
                    0 -> {
                        title = R.string.before_title
                        body = R.string.before_body
                    }
                    1 -> {
                        title = R.string.during_title
                        body = R.string.during_body
                    }
                    2 -> {
                        title = R.string.after_title
                        body = R.string.after_body
                    }
                    else -> {
                        title = R.string.after_title
                        body = R.string.after_body
                    }
                }

                return InfoFragment.newInstance(title, body)
            }

            override fun getCount(): Int = 3
        }

        mViewPager?.addOnPageChangeListener(this)

        supportActionBar?.let {
            it.navigationMode = ActionBar.NAVIGATION_MODE_TABS
            it.addTab(it.newTab().setText(R.string.before).setTabListener(this))
            it.addTab(it.newTab().setText(R.string.during).setTabListener(this))
            it.addTab(it.newTab().setText(R.string.after).setTabListener(this))
        }
    }

    override fun onTabSelected(tab: ActionBar.Tab, fragmentTransaction: FragmentTransaction) {
        mViewPager?.currentItem = tab.position
    }

    override fun onTabUnselected(tab: ActionBar.Tab, fragmentTransaction: FragmentTransaction) {

    }

    override fun onTabReselected(tab: ActionBar.Tab, fragmentTransaction: FragmentTransaction) {

    }

    override fun onPageScrolled(i: Int, v: Float, i2: Int) {

    }

    override fun onPageSelected(i: Int) {
        supportActionBar?.setSelectedNavigationItem(i)
    }

    override fun onPageScrollStateChanged(i: Int) {

    }
}
